### PR TITLE
Box:  add Overflow, OverflowX and OverflowY props

### PR DESCRIPTION
**Pull request by somethiiing**
There were a few times where I needed columnReverse for mWeb designs. so this is an update for that.

In addition, overflowX and overflowY styling to avoid using `__dangerouslySetInlineStyle` for things like rounding.

### All directions: (including the two new reverse ones)
image

### Overflow:
image

### CODE REVIEW

File: packages/syntax-core/src/Box/Box.tsx [hunk 1]
```diff
@@ -60,6 +60,13 @@ type Margin =
   | 12
   | "auto";
 type Padding = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
+type Overflow =
```
Comment (somethiiing) [inline packages/syntax-core/src/Box/Box.tsx:63]: @christianvuerings it's possible that I went a little overboard on these. probably only _really_ need `visible` `hidden` `scroll` and `inherit`, but `auto` + `initial` seemed marginally useful as well?

do you have opinions on this?

File: packages/syntax-core/src/Box/Box.tsx [hunk 2]
```diff
@@ -418,6 +453,9 @@ const Box = forwardRef<HTMLDivElement, BoxProps>(function Box(
     lgMarginEnd,
     lgMarginStart,
     lgMarginTop,
+    // Overflow
+    overflowX,
+    overflowY,
```
Comment (benjroy) [inline packages/syntax-core/src/Box/Box.tsx:458]: @somethiiing are you planning to wire in the `smOverflowX`, `lgOverflowX` ... responsive versions of these props in this PR?

Comment (somethiiing) [reply packages/syntax-core/src/Box/Box.tsx:458]: i could, but i didnt think `Overflow` was one of those props that needed a sm and lg variation. I could be wrong about this, what do others think?

File: packages/syntax-core/src/Box/Box.stories.tsx [hunk 3]
```diff
@@ -207,6 +220,46 @@ export const BackgroundColor: StoryObj<typeof Box> = {
   ),
 };
 
+export const Direction: StoryObj<typeof Box> = {
```
Comment (somethiiing) [inline packages/syntax-core/src/Box/Box.stories.tsx:223]: this story probably doesnt need to be here, but it's still fairly nice i guess

File: packages/syntax-core/src/Box/Box.tsx (hunk 1 above)
Comment (christianvuerings) [reply packages/syntax-core/src/Box/Box.tsx:64]: @somethiiing can we restrict it to:

```
hidden
scroll
auto
```

File: packages/syntax-core/src/Box/Box.tsx [hunk 4]
```diff
@@ -265,6 +272,48 @@ type BoxProps = {
    * @defaultValue "start"
    */
   justifyContent?: JustifyContent;
+  /**
+   * How box behaves when content does not fit in the box on both axes.
+   *
+   * Responsive props:
```
Comment (christianvuerings) [inline packages/syntax-core/src/Box/Box.tsx:278]: @somethiiing these are not responsive for now

**Review by christianvuerings — APPROVED** [review]

**Pull request merged**